### PR TITLE
feat: add cpp sdk doc

### PR DESCRIPTION
**Pull request by cs1137195420**
Add Cpp SDK in Casdoor SDK, Qt App in Desktop SDKs.

### CODE REVIEW

**Comment by hsluoyz** [comment]
@Selflocking @leo220yuyaodog

File: docs/how-to-connect/desktop-sdks/qt-app.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+---
+title: Qt Desktop App
+description: An Qt desktop app example for Casdoor
+keywords: [qt, sdk]
+---
+
+A [Qt desktop app example](https://github.com/casdoor/casdoor-cpp-qt-example) for Casdoor.
+
+# How to run example
```
Comment (Selflocking) [inline docs/how-to-connect/desktop-sdks/qt-app.md:9]: image
In general, we do not use h1, h1 defaults to the title and will not appear in the TOC.

File: docs/how-to-connect/desktop-sdks/qt-app.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+## Prerequisites
+
+[Qt6 sdk](https://www.qt.io/download)
+
+[OpenSSL toolkit](https://www.openssl.org/source/)
+
+## Initialization
+
+You need to init requires 7 parameters, which are all string type:
```
Comment (Selflocking) [inline docs/how-to-connect/desktop-sdks/qt-app.md:19]: do you mean:
```suggestion
You need to initialize 7 parameters, which are all string type:
```

File: docs/how-to-connect/desktop-sdks/qt-app.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+![user profile](/img/howto-desktop-qt-app-userprofile.png)
+
+You can preview the whole process by the gif image below.
+![preview gif](/img/howto-desktop-qt-app-preview.gif)
+
+# How to integrate
+
+## Open the login window
+
+```qt
```
Comment (Selflocking) [inline docs/how-to-connect/desktop-sdks/qt-app.md:59]:
```suggestion
```cpp
```

File: docs/how-to-connect/desktop-sdks/qt-app.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+// Load and display the login page of Casdoor
+m_webview->page()->load(*m_signin_url);
+m_webview->show();
+```
+
+## Listen to the open application event
+
+```qt
```
Comment (Selflocking) [inline docs/how-to-connect/desktop-sdks/qt-app.md:67]:
```suggestion
```cpp
```

File: docs/how-to-connect/desktop-sdks/qt-app.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+if(!m_tcpserver->listen(QHostAddress::LocalHost, 8080)) {
+    qDebug() << m_tcpserver->errorString();
+    close();
+}
+connect(m_tcpserver, SIGNAL(newConnection()), this, SLOT(on_tcp_connected()));
+```
+
+## Use auth code to get the user info
+
+```qt
```
Comment (Selflocking) [inline docs/how-to-connect/desktop-sdks/qt-app.md:79]:
```suggestion
```cpp
```

**Review by Selflocking — CHANGES_REQUESTED** [review]

File: docs/how-to-connect/desktop-sdks/qt-app.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+| endpoint       | Your Casdoor server host/domain                                                                         | `mainwindow.h` |
+| client_id     | The Client ID of your Casdoor application                                                               | `mainwindow.h` |
+| client_secret | The Client Secret of your Casdoor application                                                           | `mainwindow.h` |
+| certificate      | The public key for the Casdoor application's cert                                                                    | `mainwindow.h` |
+| org_name      | The name of your Casdoor organization                                                                    | `mainwindow.h` |
+| app_name      | The name of your Casdoor application                                                                    | `mainwindow.h` |
+| redirect_url  | The path of the callback URL for your Casdoor application, will be `http://localhost:8080/callback` if not provided | `mainwindow.h` |
+
+If you don't set these parameters, this project will use the [Casdoor localhost demo](http://localhost:8080) as the defult Casdoor server.
```
Comment (leo220yuyaodog) [inline docs/how-to-connect/desktop-sdks/qt-app.md:31]: Do you mean if  parameter `endpoint`  is not initialized?  These parameters are vaguely expressed and may mislead the reader.

Comment (cs1137195420) [reply docs/how-to-connect/desktop-sdks/qt-app.md:31]: > Do you mean if parameter `endpoint` is not initialized? These parameters are vaguely expressed and may mislead the reader.

Yes, that's it. I will modify the expression here later.

**Comment by hsluoyz** [comment]
@Selflocking @leo220yuyaodog plz review again

**Review by leo220yuyaodog — CHANGES_REQUESTED** [review]

**Review by Selflocking — APPROVED** [review]

**Pull request merged**